src/api/procedures/launchSto.ts

```diff
@@ -235,10 +235,16 @@ export async function prepareStorage(
     offeringPortfolioId = portfolioLikeToPortfolioId(offeringPortfolio);
   } else {
     const token = new SecurityToken({ ticker }, context);
-    const {
-      primaryIssuanceAgent: { did },
-    } = await token.details();
-    offeringPortfolioId = { did };
+    const { primaryIssuanceAgents } = await token.details();
+
+    if (primaryIssuanceAgents.length !== 1) {
+      throw new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'There is no a default Primary Issuance Agent for the given asset',
+      });
+    }
+
+    offeringPortfolioId = { did: primaryIssuanceAgents[0].did };
```

Comment:
After a talk with @VictorVicente , we decided to go ahead and make the breaking change of requiring the user to pass `offeringPortfolio` instead of making it optional (please change the interface comments to reflect this)

This means that it's no longer required to fetch the PIAs here

Reply:
Remember to commit it as a breaking change